src/python/dxpy/nextflow/ImageRef.py

```diff
@@ -118,13 +118,17 @@ def _cache(self, file_name):
             except subprocess.CalledProcessError:
                 err_exit("Failed to run a subprocess command: {}".format(cmd))
         # may need wait_on_close = True??
+        extracted_digest = self._digest
+        if not self._digest:
+            digest_cmd = "docker images --no-trunc --quiet {}".format(full_image_ref)
+            extracted_digest = subprocess.check_output(digest_cmd, shell=True).decode().strip()
         uploaded_dx_file = upload_local_file(
             filename=file_name,
             project=config["DX_PROJECT_CONTEXT_ID"],
             folder=self._caching_dir,
             name=file_name,
             parents=True,
-            properties={"image_digest": self._digest}
+            properties={"image_digest": self._digest or extracted_digest}
```

Comment:
`self._digest` is already null in this if block, so should be `extraced_digest` only

Reply:
`self._digest` might be an empty string.

Reply:
in line 121 `extracted_digest = self._digest`
so do we really have to pick `self._digest or extracted_digest` ?

Reply:
cleaned up